### PR TITLE
Php upgrade

### PR DESCRIPTION
**Pull request by sven7777**

### CODE REVIEW

File: test/AssertAnnotationTest.php [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+		$expectedAnnotation = new Annotation($id, $string, $number, $stringLength, $integerMinMax);
+
+		// when
+		$expectedAnnotation->validate();
+
+		// then
+		$this->assertTrue(true);
+	}
+
+	/** @expectedException Fliglio\Web\ValidationException */
```
Comment (benschw) [inline test/AssertAnnotationTest.php]: would it be fun to use a dataprovider to test that each annotation works?
https://phpunit.readthedocs.io/en/8.5/writing-tests-for-phpunit.html#data-providers

File: test/Client/ClientTest.php [hunk 2]
```diff
@@ -28,7 +28,54 @@ public function testError() {
...
+		$resp = $client->get('https://postman-echo.com/get', []);
+		$body = json_decode($resp->getBody(), true);
+		$headers = $resp->getHeaders();
+
+		$this->assertEquals(HTTP::STATUS_OK, $resp->getStatus());
+		$this->assertEquals('https://postman-echo.com/get', $body['url']);
+		$this->assertEquals('HTTP/1.1 200 OK', $headers[0]);
+	}
+
+	public function testGetWithHeaders() {
```
Comment (oh4real) [inline test/Client/ClientTest.php]: for consistency: "testGet_withHeaders()"?

File: test/Client/ClientTest.php [hunk 3]
```diff
@@ -28,7 +28,54 @@ public function testError() {
...
+		$resp = $client->post('https://postman-echo.com/post');
+
+		$this->assertEquals(HTTP::STATUS_OK, $resp->getStatus());
+	}
+
+	public function testDelete() {
+		$client = new BasicClient();
+		$resp = $client->delete('https://postman-echo.com/delete');
+
+		$this->assertEquals(HTTP::STATUS_OK, $resp->getStatus());
```
Comment (oh4real) [inline test/Client/ClientTest.php]: Interesting, usually returns 204 (https://restfulapi.net/http-methods/#delete) curious that postman returning 200 with an object...

File: test/Client/ClientTest.php [hunk 4]
```diff
@@ -28,7 +28,54 @@ public function testError() {
...
+		$resp = $client->put('https://postman-echo.com/put');
+
+		$this->assertEquals(HTTP::STATUS_OK, $resp->getStatus());
+	}
+
+	public function testPost() {
+		$client = new BasicClient();
+		$resp = $client->post('https://postman-echo.com/post');
+
+		$this->assertEquals(HTTP::STATUS_OK, $resp->getStatus());
```
Comment (oh4real) [inline test/Client/ClientTest.php]: Usually returns 201. Interesting that postman returns 200

**Pull request merged**